stat_var_renaming/stat_var_renaming_constants.py

```diff
@@ -0,0 +1,358 @@
+# License for the specific language governing permissions and limitations under
+# the License.
+
+"""
+  This file contains various constants and helper code to generate constants
+  that are used in the Statistical Variable renaming. 
+"""
+
+import pandas as pd
+import re
+
+def capitalizeFirst(word):
+  """ Capitalizes the first letter of a string. """
+  return word[0].upper() + word[1:]
+
+def standard_name_remapper(orig_name):
+  """ General renaming function for long strings into camel case.
+
+    Text inbetween trailing parentheses is removed. 
+    Commas, dashes, and "ands" are removed. Then string is converted into camel
+    case without and spaces present.
+   """
+  # Remove any trailing parentheses.
+  paren_start = orig_name.find("(")
+  if paren_start != -1:
+    orig_name = orig_name[:paren_start]
+
+  # Removes separating words.
+  orig_name = orig_name.replace(",", " ")
+  orig_name = orig_name.replace("-", " ")
+  orig_name = orig_name.replace("and ", "")
+  return "".join([word.capitalize() for word in orig_name.split(" ")])
+
+def _create_naics_map():
+  """ Downloads all NAICS codes across long and short form codes. """
+  # Read in list of industry topics.
+  naics_codes = pd.read_excel("https://www.census.gov/eos/www/naics/2017NAICS/2-6%20digit_2017_Codes.xlsx")
+  naics_codes = naics_codes.iloc[:, [1,2]]
+  naics_codes.columns = ['NAICSCode', 'Title']
+
+  # Replace all ranges with individual rows. E.g. 31-33 -> 31, 32, 33.
+  def range_to_array(read_code):
+    if isinstance(read_code, str) and "-" in read_code:
+      lower, upper = read_code.split("-")
+      return list(range(int(lower), int(upper) +1))
+    return read_code
+
+  naics_codes = naics_codes.dropna()
+  naics_codes['NAICSCode'] = naics_codes['NAICSCode'].apply(range_to_array)
+  naics_codes = naics_codes.explode('NAICSCode')
+
+  # Add unclassified code which is used in some statistical variables.
+  naics_codes = naics_codes.append(
+    {"NAICSCode": 99, "Title": "Nonclassifiable"}, ignore_index=True)
+
+  # Query for only two letter codes.
+  short_codes = naics_codes[naics_codes['NAICSCode'] < 100]
+  short_codes = short_codes.set_index("NAICSCode")
+  short_codes = short_codes['Title'].to_dict()
+
+  # Read in overview codes.
+  overview_codes = pd.read_csv("https://data.bls.gov/cew/doc/titles/industry/high_level_industries.csv")
+  overview_codes.columns = ["NAICSCode", "Title"]
+  overview_codes = overview_codes.set_index("NAICSCode")
+  overview_codes = overview_codes['Title'].to_dict()
+
+  # Combine the two sources of codes.
+  NAICS_MAP = {}
+  combined_codes = short_codes
+  combined_codes.update(overview_codes)
+
+  # Rename industries into camel case.
+  for code, orig_name in combined_codes.items():
+    NAICS_MAP[str(code)] = standard_name_remapper(orig_name)
+
+  # Other edge cases.
+  NAICS_MAP['00'] = 'Unclassified'
+  return NAICS_MAP
+
+NAICS_MAP = _create_naics_map()
```

Comment:
Just with the nature of its use it needs to be a global variable though. Otherwise it would either need to be always recomputed or the code refactored in an unappealing way.

Reply:
Please consider adding a todo such as

\# TODO(iancostello): Consider adding function memoization, e.g., https://book.pythontips.com/en/latest/function_caching.html, or similar technique, perhaps wrapping the map inside a class with appropriate accessor functions.